mkdocs/themes/mkdocs/base.html

```diff
@@ -59,8 +59,7 @@
     </head>
 
     <body{% if page and page.is_homepage %} class="homepage"{% endif %}>
-
-        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
+        <div class="navbar fixed-top navbar-expand-lg navbar-{% if config.theme.nav_style == "light" %}light{% else %}dark{% endif %} bg-{{ config.theme.nav_style }}">
```

Comment:
I haven't checked, but based on this change I'm assuming the `navbar-*` class can only be one of `dark` or `light`, but not `primary`. Otherwise, you would just do `navbar-{{ config.theme.nav_style }}` and avoid the `if` statement.